ql/src/java/org/apache/hadoop/hive/ql/parse/StorageFormat.java

```diff
@@ -81,11 +84,34 @@ public boolean fillStorageFormat(ASTNode child) throws SemanticException {
       }
       break;
     case HiveParser.TOK_STORAGEHANDLER:
-      storageHandler = processStorageHandler(child.getChild(0).getText());
-      if (child.getChildCount() == 2) {
-        BaseSemanticAnalyzer.readProps(
-          (ASTNode) (child.getChild(1).getChild(0)),
-          serdeProps);
+      for (int i = 0; i < child.getChildCount(); i++) {
+        ASTNode grandChild = (ASTNode) child.getChild(i);
+        switch (grandChild.getToken().getType()) {
+          case HiveParser.TOK_FILEFORMAT_GENERIC:
+            String fileFormatPropertyKey = null;
+            try {
+              HiveStorageHandler handler = HiveUtils.getStorageHandler(conf, this.storageHandler);
+              fileFormatPropertyKey = handler.getFileFormatPropertyKey();
+            } catch (HiveException e) {
+              throw new SemanticException("Failed to load storage handler:  " + e.getMessage());
+            }
+
+            if (fileFormatPropertyKey != null) {
+              String fileFormat = grandChild.getChild(0).getText();
+              if (serdeProps.containsKey(fileFormatPropertyKey)) {
+                throw new SemanticException("Provide only one of the following: STORED BY " + fileFormat +
+                    " or WITH SERDEPROPERTIES('" + fileFormatPropertyKey + "'='" + fileFormat + "')");
+              }
+
+              serdeProps.put(fileFormatPropertyKey, fileFormat);
```

Comment:
What happens if the fileformat is Text or some other valid  fileformat which is not supported by Iceberg?

Reply:
If the provided file format is not supported, iceberg defaults back to `avro`.  After the table creation, the metadata shows the not supported file format, but right after the first insert the property value is updated to `avro`.